posthog/async_migrations/runner.py

```diff
@@ -173,15 +170,9 @@ def attempt_migration_rollback(migration_instance: AsyncMigration, force: bool =
     try:
         ops = get_async_migration_definition(migration_instance.name).operations
 
-        i = len(ops) - 1
-        for op in reversed(ops[: migration_instance.current_operation_index + 1]):
-            if not op.rollback:
-                if op.rollback == "":
-                    continue
-                raise Exception(f"No rollback provided for operation at index {i}: {op.sql}")
+        for op_index in range(migration_instance.current_operation_index, -1, -1):
+            op = ops[op_index]
             execute_op(op, str(UUIDT()), rollback=True)
-            op.side_effect_rollback()
```

Comment:
Note that we also did this in a bad location before (this pr fixes that). E.g. 0002 last step would detach table before turning compute materialized columns off but we should do that in reverse order.